jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -572,18 +573,31 @@ private IGuiMovable getSelectableMovable(short x, short y) {
 
 	private void selectPointType(PointAction action) {
 		final ShortPoint2D actionPosition = action.getPosition();
-		final IGuiMovable centerSelectable = getSelectableMovable(actionPosition.x, actionPosition.y);
+		final IGuiMovable selectedMovable = getSelectableMovable(actionPosition.x, actionPosition.y);
 
-		if (centerSelectable == null) { // nothing found at the location
+		if (selectedMovable == null) { // nothing found at the location
 			setSelection(new SelectionSet());
 			return;
 		}
 
-		final List<ISelectable> selected = new LinkedList<ISelectable>();
-	
+		EMovableType selectedType = selectedMovable.getMovableType();
+
+		Set<EMovableType> selectableTypes;
+		if (selectedType.isSwordsman()) {
```

Comment:
Yes, you would and you can still leave the else part. This should just replace the isPikeman()..

You can then get the movable set from the ESoldierType.